### PR TITLE
Delete can now delete errored instances.

### PR DESCRIPTION
**Pull request by ed-**
Let's try this again, a little more organized now.

### CODE REVIEW

File: reddwarf/instance/models.py [hunk 1]
```diff
@@ -305,6 +316,25 @@ def __init__(self, context, db_info, server, service_status):
     def get_guest(self):
         return create_guest_client(self.context, self.db_info.id)
 
+    def _delete_resources(self):
+        pass
+
+    def delete_instance(self):
+        # This is here to be passed down to BuiltInstanceTasks and
```
Comment (TimSimpsonR) [inline reddwarf/instance/models.py]: NGHUTPR of course but this comment isn't really true. This code makes sense from the standpoint of any instance- it should work for even the basest of the instance base classes.

**Comment by TimSimpsonR** [comment]
+1

**Comment by imsplitbit** [comment]
+1

File: reddwarf/instance/models.py [hunk 2]
```diff
@@ -305,6 +316,33 @@ def __init__(self, context, db_info, server, service_status):
     def get_guest(self):
         return create_guest_client(self.context, self.db_info.id)
 
+    def delete(self):
+        if (self.db_info.server_status in ["BUILD"] and
```
Comment (cp16net) [inline reddwarf/instance/models.py:320]: we have now avoided having an instance stay in a build state forever i assume? I think this is the right approach but just curious when was the last time we saw build forever?

**Comment by jcru** [comment]
+1

**Comment by cp16net** [comment]
+1

**Pull request merged**